src/main/java/org/broadinstitute/hellbender/tools/examples/ExampleAssemblyRegionWalker.java

```diff
@@ -45,12 +42,7 @@ public AssemblyRegionEvaluator assemblyRegionEvaluator() {
 
     @Override
     public void onTraversalStart() {
-        try {
-            outputStream = outputFile != null ? new PrintStream(outputFile) : System.out;
-        }
-        catch ( final FileNotFoundException e ) {
-            throw new UserException.CouldNotCreateOutputFile(outputFile, e);
-        }
+        outputStream = outputFile != null ? new PrintStream(outputFile.getOutputStream()) : System.out;
```

Comment:
The `PrintStream` constructor can still end up throwing an exception, which we should handle (especially in an example tool like this). If you look at the implementation of `PrintStream`, you can see that the constructor ultimately ends up creating an `OutputStreamWriter`, which can throw...

Reply:
I added back in a catch block for `Exception` since there are no checked exceptions thrown (though that won't catch the `Error` explicitly thrown by OutputStreamWriter).

Reply:
Can/should we explicitly catch the `Error` here (and in other places where `OutputStreamWriter` is used)?

Reply:
I thought catching `Error` was generally discouraged, so its seems like bad form, especially in sample code. I think if we really want to do it right we should eliminate the use of  PrintStream - those classes seem sketchy around.

Reply:
@cmnbroad I think it's ok to catch `Error` just to wrap it in a nicer exception type (ie., `UserException`). I believe what's discouraged is catching Error and then attempting to recover from it.